worlds/tloz/__init__.py

```diff
@@ -0,0 +1,308 @@
+        for i in range(1, 10):
+            level = Region(f"Level {i}", self.player, self.multiworld)
+            self.levels.append(level)
+            new_entrance = Entrance(self.player, f"Level {i}", overworld)
+            new_entrance.connect(level)
+            overworld.exits.append(new_entrance)
+            self.multiworld.regions.append(level)
+
+        for i, level in enumerate(level_locations):
+            for location in level:
+                if self.multiworld.ExpandedPool[self.player] or "Drop" not in location:
+                    self.levels[i + 1].locations.append(
+                        self.create_location(location, self.location_name_to_id[location], self.levels[i + 1]))
+
+        for level in range(1, 9):
+            boss_event = self.create_location(f"Level {level} Boss Status", None,
+                                              self.multiworld.get_region(f"Level {level}", self.player),
+                                              True)
+            boss_event.show_in_spoiler = False
+            self.levels[level].locations.append(boss_event)
+
+        for location in major_locations:
+            if self.multiworld.ExpandedPool[self.player] or "Take Any" not in location:
+                overworld.locations.append(
+                    self.create_location(location, self.location_name_to_id[location], overworld))
+
+        for location in shop_locations:
+            overworld.locations.append(
+                self.create_location(location, self.location_name_to_id[location], overworld))
+
+        ganon = self.create_location("Ganon", None, self.multiworld.get_region("Level 9", self.player))
+        zelda = self.create_location("Zelda", None, self.multiworld.get_region("Level 9", self.player))
+        ganon.show_in_spoiler = False
+        zelda.show_in_spoiler = False
+        self.levels[9].locations.append(ganon)
+        self.levels[9].locations.append(zelda)
+        begin_game = Entrance(self.player, "Begin Game", menu)
+        menu.exits.append(begin_game)
+        begin_game.connect(overworld)
+        self.multiworld.regions.append(menu)
+        self.multiworld.regions.append(overworld)
+
+    set_rules = set_rules
+
+    def generate_basic(self):
+        ganon = self.multiworld.get_location("Ganon", self.player)
+        ganon.place_locked_item(self.create_event("Triforce of Power"))
+        add_rule(ganon, lambda state: state.has("Silver Arrow", self.player) and state.has("Bow", self.player))
+
+        self.multiworld.get_location("Zelda", self.player).place_locked_item(self.create_event("Rescued Zelda!"))
+        add_rule(self.multiworld.get_location("Zelda", self.player),
+                 lambda state: ganon in state.locations_checked)
+
+        self.multiworld.completion_condition[self.player] = lambda state: state.has("Rescued Zelda!", self.player)
+        generate_itempool(self)
```

Comment:
create items in the function dedicated to it please.

Reply:
since all of the item creation is done in this function that is in a whole other file (ItemPool.py), sticking it in create_items seemed redundant to me? unless theres something fundamental im not understanding

Reply:
You're doing it later than expected, which has lead to bugs before and in the future AP is likely to reject your world for it.